modules/akka-downing/src/main/scala/ch/epfl/bluebrain/nexus/commons/downing/KeepOldestAkkaDowningProvider.scala

```diff
@@ -0,0 +1,148 @@
+package ch.epfl.bluebrain.nexus.commons.downing
+
+import akka.actor.{Actor, ActorLogging, ActorSystem, Cancellable, Props}
+import akka.cluster.ClusterEvent._
+import akka.cluster.{Cluster, DowningProvider, Member, MemberStatus, UniqueAddress}
+import ch.epfl.bluebrain.nexus.commons.downing.config.{DowningConfig, Settings}
+
+import scala.concurrent.duration._
+
+class KeepOldestAkkaDowningProvider(system: ActorSystem) extends DowningProvider {
```

Comment:
Please add a header explaining the global logic for this provider and its variations.